posthog-node/src/client.ts

```diff
@@ -686,6 +730,34 @@ export abstract class PostHogBackendClient extends PostHogCoreStateless implemen
     }
   }
 
+  private extractPropertiesFromEvent(eventProperties?: Record<string | number, any>): {
```

Comment:
I think this method should also accept `groups: EventMessage['groups']` and only extract groups from `eventProperties` when the key is in `groups`. Does that make sense?

Reply:
Yeah that's a good idea.  I was basically defaulting to keeping the data extraction dumb in the client (since the server will handle mismatches of properties), but I can be more hygienic in this case.  I'll change.